Harvest.Core/Services/InvoiceService.cs

```diff
@@ -232,5 +234,29 @@ public async Task<List<int>> GetCreatedInvoiceIds()
             return await _dbContext.Invoices.Where(a => a.Status == Invoice.Statuses.Created).Select(a => a.Id)
                 .ToListAsync();
         }
+
+        /// <summary>
+        /// Automatically closeout projects that are 18 days or older since closeout was initiated.
+        /// </summary>
+        /// <param name="daysBeforeAutoCloseout">Negative Value</param>
+        /// <returns></returns>
+        public async Task<int> AutoCloseoutProjects(int daysBeforeAutoCloseout)
+        {
+            var dateCutoff = DateTime.UtcNow.AddDays(daysBeforeAutoCloseout);
+
+            var count = 0;
+            var potentialProjectIds = await _dbContext.Projects.Where(a => a.IsActive && a.Status == Project.Statuses.PendingCloseoutApproval).Select(a => a.Id).ToArrayAsync();
+            var projectIds = await _dbContext.ProjectHistory.Where(a => a.Action == "ProjectCloseoutInitiated" && potentialProjectIds.Contains(a.ProjectId) && a.ActionDate < dateCutoff).Select(a => a.ProjectId).ToArrayAsync();
+
+            foreach (var projectId in projectIds)
+            {
+                var result = await CreateInvoice(projectId, true);

```

Comment:
let's add some logging since this is a fairly sensitive thing, auto closeout.  also maybe a try/catch in case one project has issues

Reply:
@jSylvestre yea i was just thinking something simple like add a specific log saying something like "Initiating auto closeout for project 123"